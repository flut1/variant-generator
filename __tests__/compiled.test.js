import createVariants, { iterateCombinations } from '../index';

describe('compiled js', () => {

  describe('createVariants()', () => {
    test('handles simple 2x2 input', () => {
      const result = createVariants(function*() {
        return {
          bool: yield [false, true],
          num: yield [0, 1],
        };
      });

      expect(Array.from(result)).toMatchSnapshot();
    });
  });

  describe('iterateCombinations', () => {
    test('it runs once for each combination of parameters', () => {
      const spy = jest.fn();

      iterateCombinations(function*() {
        yield [1, 2, 3];

        yield [4, 5, 6];

        yield [7, 8];

        spy();
      });

      expect(spy).toBeCalledTimes(3 * 3 * 2);
    });
  });

  test('handles conditional yields', () => {
    const result = createVariants(function*() {
      const fruit = yield ['banana', 'apple', 'strawberry'];
      let size = 'normal';

      switch (fruit) {
        case 'banana':
          size = yield ['small', 'normal', 'large'];
          break;
        case 'apple':
          size = yield ['small', 'normal'];
          break;
      }
      let color = 'red';

      switch (fruit) {
        case 'banana':
          color = yield ['yellow', 'green'];
          break;
        case 'apple':
          color = yield ['green', 'red'];
          break;
      }

      return {
        fruit,
        size,
        color,
      };
    });

    expect(Array.from(result)).toMatchSnapshot();
  });
});
