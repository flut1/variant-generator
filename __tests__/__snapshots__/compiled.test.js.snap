// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compiled js createVariants() handles simple 2x2 input 1`] = `
Array [
  Object {
    "bool": false,
    "num": 0,
  },
  Object {
    "bool": false,
    "num": 1,
  },
  Object {
    "bool": true,
    "num": 0,
  },
  Object {
    "bool": true,
    "num": 1,
  },
]
`;

exports[`compiled js handles conditional yields 1`] = `
Array [
  Object {
    "color": "yellow",
    "fruit": "banana",
    "size": "small",
  },
  Object {
    "color": "green",
    "fruit": "banana",
    "size": "small",
  },
  Object {
    "color": "yellow",
    "fruit": "banana",
    "size": "normal",
  },
  Object {
    "color": "green",
    "fruit": "banana",
    "size": "normal",
  },
  Object {
    "color": "yellow",
    "fruit": "banana",
    "size": "large",
  },
  Object {
    "color": "green",
    "fruit": "banana",
    "size": "large",
  },
  Object {
    "color": "green",
    "fruit": "apple",
    "size": "small",
  },
  Object {
    "color": "red",
    "fruit": "apple",
    "size": "small",
  },
  Object {
    "color": "green",
    "fruit": "apple",
    "size": "normal",
  },
  Object {
    "color": "red",
    "fruit": "apple",
    "size": "normal",
  },
  Object {
    "color": "red",
    "fruit": "strawberry",
    "size": "normal",
  },
]
`;
